### PR TITLE
[SPARK-26212][Build][test-maven] Upgrade maven version to 3.6.0

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR updates maven version from 3.5.4 to 3.6.0. The release note of the 3.6.0 is [here](https://maven.apache.org/docs/3.6.0/release-notes.html).

From [the release note of the 3.6.0](https://maven.apache.org/docs/3.6.0/release-notes.html), the followings are new features:
1. There had been issues related to the project discoverytime which has been increased in previous version which influenced some of our users.
1. The output in the reactor summary has been improved.
1. There was an issue related to the classpath ordering.

## How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by kiszk** [comment]
retest this please

**Comment by srowen** [comment]
That's fine but we need to also update build/mvn to download and use the same version.

**Comment by kiszk** [comment]
I thought that it is automatically done by `build/mvn`, as you pointed out [before](https://github.com/apache/spark/pull/21905#issuecomment-408678119).

**Comment by srowen** [comment]
Ah, that's the second time I've forgotten this. Yes looks good to me.

File: pom.xml [hunk 1]
```diff
@@ -114,7 +114,7 @@
     <java.version>1.8</java.version>
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
-    <maven.version>3.5.4</maven.version>
+    <maven.version>3.6.0</maven.version>
```
[the remark this answers is not in the document]
Comment (srowen) [reply pom.xml]: That wouldn't hurt, though I suspect the project continues to build with Maven 3.5.4 for a long while.

**Comment by dongjoon-hyun** [comment]
Thank you for pinging me, @HyukjinKwon .

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply pom.xml]: +1 for updating the doc, too.

**Review by dongjoon-hyun — COMMENTED** [review]
LGTM except for the doc issue.
I also tested with Scala-2.11 Maven for safety. It looks okay.

**Comment by dongjoon-hyun** [comment]
Oh, @kiszk . Could you update AppVoyer script together? AppVoyer still uses a static version number unlike with `build/mvn`.
- https://github.com/apache/spark/blob/master/dev/appveyor-install-dependencies.ps1#L84

**Comment by kiszk** [comment]
Sure, updated. Thanks for letting know them.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
Merging to master. I've been using 3.6.0 on the command line for a while and it's fine. Note that if you use your local mvn in IntelliJ, it seems to have some incompatibility with the current latest 2018.13.1 release. It's no big deal, falling back to its internal 3.3.9 version works fine.

**Pull request closed without merging**